components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/APIMappingUtil.java

```diff
@@ -814,6 +814,22 @@ static APIInfoDTO fromAPIToInfoDTO(API api) throws APIManagementException {
                 
             }
         }
+        if (api.getAdditionalProperties() != null) {
+            JSONObject additionalProperties = api.getAdditionalProperties();
+            List<APIInfoAdditionalPropertiesDTO> additionalPropertiesList = new ArrayList<>();
+            for (Object propertyKey : additionalProperties.keySet()) {
+                APIInfoAdditionalPropertiesDTO additionalPropertiesDTO = new APIInfoAdditionalPropertiesDTO();
+                String key = (String) propertyKey;
+                int index = key.lastIndexOf(APIConstants.API_RELATED_CUSTOM_PROPERTIES_SURFIX);
+                additionalPropertiesDTO.setValue((String) additionalProperties.get(key));
+                if (index > 0) {
+                    additionalPropertiesDTO.setName(key.substring(0, index));
+                    additionalPropertiesDTO.setDisplay(true);
```

Comment:
when adding an additional property from publisher API, we set whether display is true or false. Do we always get the properties whose display value is true here? Otherwise we need to filter them out.

Reply:
Yes. Its filtered by `int index = key.lastIndexOf(APIConstants.API_RELATED_CUSTOM_PROPERTIES_SURFIX);` which gets whether the display prop is set or not.